src/Domain/FunctionsOfTime/QuaternionFunctionOfTime.cpp

```diff
@@ -126,6 +126,8 @@ void QuaternionFunctionOfTime<MaxDeriv>::pup(PUP::er& p) {
 
       // Same for v0 and v1
       p | angle_f_of_t_;
+      expiration_time_.store(angle_f_of_t_.time_bounds()[1],
+                             std::memory_order_relaxed);
```

Comment:
For this file and `PiecewisePolynomial.cpp` I wanted to always use `std::memory_order_{acquire/release}` rather than `std::memory_order_relaxed` (even though in `pup` it doesn't really make a difference). However, I realize I missed a spot a several lines below which is presumably why you used it here, so that's my bad. I'll update this in a follow-up PR.